### PR TITLE
Bug fix habit name checking and standardize error

### PR DESCRIPTION
**Pull request by BernardLesley**
Summary of Bug Fix:
1. Add checking to prevent habit name containing only numbers and symbol
2. Preserve order of habit
3. Standardise tokenizer error
5. Overwrite erroneous data in atomic habit to prevent recurring error message

### CODE REVIEW

**Comment by BernardLesley** [comment]
This closes #328  and help to resolve #318

File: src/main/java/wellnus/reflection/feature/QuestionList.java [hunk 1]
```diff
@@ -78,8 +78,8 @@ public QuestionList() {
         try {
             storage = new Storage();
         } catch (StorageException storageException) {
-            LOGGER.log(Level.WARNING, STORAGE_ERROR);
-            UI.printErrorFor(storageException, STORAGE_ERROR);
+            LOGGER.log(Level.WARNING, TOKENIZER_ERROR);
+            UI.printErrorFor(storageException, TOKENIZER_ERROR);
```
Comment (wenxin-c) [inline src/main/java/wellnus/reflection/feature/QuestionList.java:82]: should this be storage error instead?

**Review by wenxin-c — CHANGES_REQUESTED** [review]
Minor error need to change

**Review by wenxin-c — COMMENTED** [review]
Error fixed

File: src/main/java/wellnus/storage/ReflectionTokenizer.java [hunk 2]
```diff
@@ -26,9 +24,8 @@ public class ReflectionTokenizer implements Tokenizer<Set<Integer>> {
     private static final String PREV_KEY = "prev";
     private static final String COLON_CHARACTER = ":";
     private static final int NO_LIMIT = -1;
-    private static final String FILE_NAME = "reflect";
-    private static final String DETOKENIZE_ERROR_MESSAGE = "Detokenization failed! "
-            + "The file might be corrupted!";
+    private static final String DETOKENIZE_ERROR_MESSAGE = "Invalid reflect %s data '%s' found in storage!";
+    private static final String STORAGE_ERROR = "Previous reflect %s data will not be restored.";
```
Comment (haoyangw) [inline src/main/java/wellnus/storage/ReflectionTokenizer.java:28]: Unused variable i believe

**Review by haoyangw — APPROVED** [review]
Changes seem to work as intended, no code style issues here either. Apart from an unused variable, LGTM!

**Review by wenxin-c — APPROVED** [review]
LGTM!!

**Pull request merged**